crates/synthesis/src/linalg/mod.rs

```diff
@@ -12,15 +12,60 @@
 
 use approx::{abs_diff_eq, relative_ne};
 use faer::MatRef;
-use faer_ext::{IntoFaer, IntoNalgebra};
-use nalgebra::{DMatrix, DMatrixView};
+use faer_ext::IntoFaer;
+use nalgebra::{DMatrix, DMatrixView, Dim, Dyn, MatrixView, ViewStorage};
+use ndarray::ArrayView2;
+use ndarray::ShapeBuilder;
 use num_complex::Complex64;
 
 pub mod cos_sin_decomp;
 
 const ATOL_DEFAULT: f64 = 1e-8;
 const RTOL_DEFAULT: f64 = 1e-5;
 
+fn nalgebra_to_faer<R: Dim, C: Dim, RStride: Dim, CStride: Dim>(
+    mat: MatrixView<Complex64, R, C, RStride, CStride>,
+) -> MatRef<Complex64> {
+    let dim = ::ndarray::Dim(mat.shape());
+    let strides = ::ndarray::Dim(mat.strides());
+
+    // SAFETY: We know the array is a 2x2 and contiguous block (DMatrix uses a vec for backing        storage) so we
```

Comment:
How do we know it's 2x2? Did you mean 2d?

Reply:
Sigh, yeah I meant 2d. I copied this from a different place where I did something similar but where it was a `Matrix2` not a `DMatrix`.

```suggestion
    // SAFETY: We know the array is a 2d contiguous block (DMatrix uses a vec for backing storage) so we
```